service/serialize.go

```diff
@@ -0,0 +1,153 @@
+package service
+
+import (
+	"strconv"
+
+	"github.com/mesg-foundation/engine/hash/hashserializer"
+)
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Service) HashSerialize() string {
+	if data == nil {
```

Comment:
Not a discussion for this PR but we might simplify the hash for the service in order to have similar services producing the same hash to only
`Name + Tasks + Events + Dependencies + Sid + Source`
Even further, the source is already a hash from IPFS so we could have only this in the hash (but I like to keep the definition too)

Reply:
https://github.com/mesg-foundation/engine/issues/1709